src/main/java/com/okta/sdk/framework/RateLimitContext.java

```diff
@@ -0,0 +1,58 @@
+package com.okta.sdk.framework;
+
+import com.okta.sdk.exceptions.SdkException;
+import org.apache.http.Header;
+import org.apache.http.HttpResponse;
+import org.joda.time.DateTime;
+
+public class RateLimitContext {
+    private HttpResponse httpResponse;
+
+    public RateLimitContext(HttpResponse httpResponse) {
+        this.httpResponse = httpResponse;
+    }
+
+    private String getHeaderValueString(String headerName) throws Exception {
+        if (httpResponse == null) {
+            throw new SdkException("No http response");
+        }
+
+        Header[] headers = httpResponse.getHeaders(headerName);
+        if (headers.length > 0) {
+            Header header = headers[0];
+            return header.getValue();
+        } else {
+            throw new SdkException("No " + headerName + " header");
+        }
+    }
+
+    private Long getHeaderValueLong(String headerName) throws Exception {
```

Comment:
Would be better to have all these methods return a primitive `long` rather than an Object `Long`.